src/main/scala/analysis/GlobalRegionAnalysis.scala

```diff
@@ -0,0 +1,246 @@
+                           val mmm: MemoryModelMap,
+                           val globalOffsets: Map[BigInt, BigInt],
+                           val vsaResult: Option[Map[CFGPosition, LiftedElement[Map[Variable | MemoryRegion, Set[Value]]]]]) {
+
+  var dataCount: Int = 0
+  private def nextDataCount() = {
+    dataCount += 1
+    s"data_$dataCount"
+  }
+
+  val regionLattice: PowersetLattice[DataRegion] = PowersetLattice()
+
+  val lattice: MapLattice[CFGPosition, Set[DataRegion], PowersetLattice[DataRegion]] = MapLattice(regionLattice)
+
+  val first: Set[CFGPosition] = Set.empty + program.mainProcedure
+
+  private val stackPointer = Register("R31", 64)
+  private val linkRegister = Register("R30", 64)
+  private val framePointer = Register("R29", 64)
+  private val mallocVariable = Register("R0", 64)
+
+  private val dataMap: mutable.HashMap[BigInt, DataRegion] = mutable.HashMap()
+
+  private def dataPoolMaster(offset: BigInt, size: BigInt): Option[DataRegion] = {
+    assert(size >= 0)
+    if (dataMap.contains(offset)) {
+      if (dataMap(offset).size < (size.toDouble / 8).ceil.toInt) {
+        dataMap(offset) = DataRegion(dataMap(offset).regionIdentifier, offset, (size.toDouble / 8).ceil.toInt)
+        Some(dataMap(offset))
+      } else {
+        Some(dataMap(offset))
+      }
+    } else {
+      dataMap(offset) = DataRegion(nextDataCount(), offset, (size.toDouble / 8).ceil.toInt)
+      Some(dataMap(offset))
+    }
+  }
+
+  def getDataMap: mutable.HashMap[BigInt, DataRegion] = dataMap
+
+  def resolveGlobalOffsetSecondLast(address: BigInt): BigInt = {
+    var tableAddress = address
+    // addresses may be layered as in jumptable2 example for which recursive search is required
+    var exitLoop = false
+    while (globalOffsets.contains(tableAddress) && globalOffsets.contains(globalOffsets(tableAddress)) && !exitLoop) {
+      val newAddress = globalOffsets.getOrElse(tableAddress, tableAddress)
+      if (newAddress == tableAddress) {
+        exitLoop = true
+      } else {
+        tableAddress = newAddress
+      }
+    }
+    tableAddress
+  }
+
+  def tryCoerceIntoData(exp: Expr, n: Command, subAccess: BigInt): Set[DataRegion] = {
+    val eval = evaluateExpression(exp, constantProp(n))
+    if (eval.isDefined) {
+      val region = dataPoolMaster(eval.get.value, subAccess)
+      if (region.isDefined) {
+          return Set(region.get)
+      }
+    }
+    exp match
+      case literal: BitVecLiteral => tryCoerceIntoData(literal, n, subAccess)
+      case Extract(end, start, body) => tryCoerceIntoData(body, n, subAccess)
+      case Repeat(repeats, body) => tryCoerceIntoData(body, n, subAccess)
+      case ZeroExtend(extension, body) => tryCoerceIntoData(body, n, subAccess)
+      case SignExtend(extension, body) => tryCoerceIntoData(body, n, subAccess)
+      case UnaryExpr(op, arg) => tryCoerceIntoData(arg, n, subAccess)
```

Comment:
None of these are robust ways of trying to identify which data regions an expression points to. You can't just ignore these operations like this, it will give incorrect results.